### PR TITLE
Add support for Heatit Z-Water TF 20a

### PR DESCRIPTION
**Pull request by larstobi**

### CODE REVIEW

File: config/thermofloor/heatit20a.xml [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    <MetaDataItem id="020a" name="FrequencyName" type="0003">CEPT (Europe)</MetaDataItem>
+    <MetaDataItem name="ResetDescription">By pressing buttons Right and Center (down and confirm) for 20 seconds, the thermostat will perform a complete factory reset.
+NB! Please use this procedure only when the primary controller/ gateway is missing or otherwise inoperable.</MetaDataItem>
+    <MetaDataItem name="Name">Heatit Z-Water</MetaDataItem>
+    <ChangeLog>
+      <Entry author="Espen Throndsen - espenthrondsen@gmail.com" date="21 Oct 2019" revision="1">Initial Metadata Import from Z-Wave Alliance Database - https://products.z-wavealliance.org/products/3102/xml</Entry>
+      <Entry author="Lars Tobias Skjong-Børsting - larstobi@relatime.no" date="28 Jun 2020" revision="1">Added ProductPic</Entry>
+    </ChangeLog>
+  </MetaData>
+	<CommandClass id="32" name="COMMAND_CLASS_BASIC" version="1" request_flags="5" mapping="37">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:32]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+				<Instance index="4" endpoint="3" />
+				<Instance index="5" endpoint="4" />
+				<Instance index="6" endpoint="5" />
+				<Instance index="7" endpoint="6" />
+				<Instance index="8" endpoint="7" />
+				<Instance index="9" endpoint="8" />
+				<Instance index="10" endpoint="9" />
+				<Instance index="11" endpoint="10" />
+	</CommandClass>
+	<CommandClass id="37" name="COMMAND_CLASS_SWITCH_BINARY" version="1" request_flags="5" issecured="true" innif="true">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:45]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+				<Value type="bool" genre="user" instance="7" index="0" label="Switch 06" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="False" />
+				<Value type="bool" genre="user" instance="8" index="0" label="Switch 07" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="False" />
+				<Value type="bool" genre="user" instance="9" index="0" label="Switch 08" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="False" />
+				<Value type="bool" genre="user" instance="10" index="0" label="Switch 09" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="False" />
+				<Value type="bool" genre="user" instance="11" index="0" label="Switch 10" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="False" />
+	</CommandClass>
+	<CommandClass id="38" name="COMMAND_CLASS_SWITCH_MULTILEVEL" version="1">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:69]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+				<Instance index="1" />
+				<Value type="byte" genre="user" instance="1" index="0" label="Level" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="255" value="0" />
+				<Value type="button" genre="user" instance="1" index="1" label="Bright" units="" read_only="false" write_only="true" verify_changes="false" poll_intensity="0" min="0" max="0" />
+				<Value type="button" genre="user" instance="1" index="2" label="Dim" units="" read_only="false" write_only="true" verify_changes="false" poll_intensity="0" min="0" max="0" />
+				<Value type="bool" genre="system" instance="1" index="3" label="Ignore Start Level" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="True" />
+				<Value type="byte" genre="system" instance="1" index="4" label="Start Level" units="" read_only="false" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="255" value="0" />
+	</CommandClass>
+	<CommandClass id="49" name="COMMAND_CLASS_SENSOR_MULTILEVEL" version="1">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:77]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+				<Instance index="1" endpoint="11" />
+				<Instance index="2" endpoint="12" />
+				<Instance index="3" endpoint="13" />
+				<Instance index="4" endpoint="14" />
+				<Value type="decimal" genre="user" instance="1" index="1" label="Temperature input 1" units="C" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="-32.25" />
+				<Value type="decimal" genre="user" instance="2" index="1" label="Temperature input 1" units="C" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="-32.25" />
+				<Value type="decimal" genre="user" instance="3" index="1" label="Temperature input 1" units="C" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="-31.92" />
+				<Value type="decimal" genre="user" instance="4" index="1" label="Temperature input 1" units="C" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="0" value="-32.25" />
+	</CommandClass>
+	<CommandClass id="90" name="COMMAND_CLASS_DEVICE_RESET_LOCALLY" version="1" request_flags="5" issecured="true" innif="true">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:87]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+				<Instance index="1" />
+	</CommandClass>
+	<CommandClass id="94" name="COMMAND_CLASS_ZWAVEPLUS_INFO" version="1" request_flags="5" innif="true">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:90]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+				<Instance index="1" />
+				<Value type="byte" genre="system" instance="1" index="0" label="ZWave+ Version" units="" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="0" max="255" value="1" />
+				<Value type="short" genre="system" instance="1" index="1" label="InstallerIcon" units="" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="-32768" max="32767" value="5376" />
+				<Value type="short" genre="system" instance="1" index="2" label="UserIcon" units="" read_only="true" write_only="false" verify_changes="false" poll_intensity="0" min="-32768" max="32767" value="5376" />
+	</CommandClass>
+	<CommandClass id="114" name="COMMAND_CLASS_MANUFACTURER_SPECIFIC" version="1" request_flags="5" issecured="true" innif="true">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:96]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+				<Instance index="1" />
+	</CommandClass>
+	<CommandClass id="115" name="COMMAND_CLASS_POWERLEVEL" version="1" request_flags="5" issecured="true" innif="true">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:99]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+      <Help>This parameter decides which kind of thermistor that is connected to the input.</Help>
+      <Item label="0 No thermistor, input is disabled. " value="0"/>
+      <Item label="1 10K NTC. (PART NUMBER: TT02-10KC3-93D-3000R-TPH). (Default)" value="1"/>
+    </Value>
+    <Value genre="config" index="6" instance="1" label="Thermistor type connected to input 4." size="1" type="list" value="0">
+      <Help>This parameter decides which kind of thermistor that is connected to the input.</Help>
+      <Item label="0 No thermistor, input is disabled. " value="0"/>
+      <Item label="1 10K NTC. (PART NUMBER: TT02-10KC3-93D-3000R-TPH). (Default)" value="1"/>
+    </Value>
+    <Value genre="config" index="7" instance="1" label=" Input 1 calibration." size="1" type="byte"  max="255" min="0" size="1" type="byte" value="0">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:169]: you have two size and byte entries.

File: config/thermofloor/heatit20a.xml [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+      <Help>-40 - 40 (-4.0C - 4.0C) Default is 0 (0.0C) To set a negative value, use 255 and subtract the desired value.</Help>
+    </Value>
+    <Value genre="config" index="8" instance="1" label=" Input 2 calibration." size="1" type="byte"  max="255" min="0" size="1" type="byte" value="0">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:172]: same as above.

File: config/thermofloor/heatit20a.xml [hunk 11]
```diff
@@ -0,0 +1,225 @@
...
+      <Help>-40 - 40 (-4.0C - 4.0C) Default is 0 (0.0C) To set a negative value, use 255 and subtract the desired value.</Help>
+    </Value>
+    <Value genre="config" index="9" instance="1" label=" Input 3 calibration." size="1" type="byte"  max="255" min="0" size="1" type="byte" value="0">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:175]: same as above.

File: config/thermofloor/heatit20a.xml [hunk 12]
```diff
@@ -0,0 +1,225 @@
...
+      <Help>-40 - 40 (-4.0C - 4.0C) Default is 0 (0.0C) To set a negative value, use 255 and subtract the desired value.</Help>
+    </Value>
+    <Value genre="config" index="10" instance="1" label=" Input 4 calibration." size="1" type="byte"  max="255" min="0" size="1" type="byte" value="0">
```
Comment (Fishwaldo) [inline config/thermofloor/heatit20a.xml:178]: This should not be required. It should be auto discovered.

File: config/thermofloor/heatit20a.xml (hunk 1 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Removed

File: config/thermofloor/heatit20a.xml (hunk 2 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Removed

File: config/thermofloor/heatit20a.xml (hunk 3 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Removed

File: config/thermofloor/heatit20a.xml (hunk 4 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Removed

File: config/thermofloor/heatit20a.xml (hunk 5 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Removed

File: config/thermofloor/heatit20a.xml (hunk 6 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Removed

File: config/thermofloor/heatit20a.xml (hunk 7 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Removed

File: config/thermofloor/heatit20a.xml (hunk 8 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml:99]: Removed

File: config/thermofloor/heatit20a.xml (hunk 9 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Fixed

File: config/thermofloor/heatit20a.xml (hunk 10 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Fixed

File: config/thermofloor/heatit20a.xml (hunk 11 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Fixed

File: config/thermofloor/heatit20a.xml (hunk 12 above)
Comment (larstobi) [reply config/thermofloor/heatit20a.xml]: Do you mean only this particular line, or the entire "  <CommandClass id="112">" ?

**Comment by larstobi** [comment]
Thanks for you feedback! 

I have tried to do as you requested, but I wasn't sure about the last comment.

**Comment by Fishwaldo** [comment]
Fixed up a few more things. Thanks!

File: config/thermofloor/heatit20a.xml (hunk 3 above)
Comment (fredrikz) [reply config/thermofloor/heatit20a.xml]: @larstobi Did you ever get this to work? I've bought the HeatIt Z Water unit, and the 10 binary switches for the relays is not auto-discovered. Adding similar lines as above makes them appear in ozwadmin/Home Assistant, but I'm very new to all this and not sure if that's the way to do it. Without adding a line for each binary switch, only a single binary switch is reported, which toggles all 10 relays simultaneously, which is not ideal as it defeats the purpose of this device.

@Fishwaldo What can make this auto-discovery fail? Any setting on my end that can be changed? Running OpenZWave as an addon to for Home Assistant, based on the commit 66619eda6caaf2b41d07d68672efa7170f6c4722

**Pull request merged**